### PR TITLE
add Dfsioe Bench

### PR DESCRIPTION
**Pull request by Moonlit-Sailor**
Add dfsioe bench to Hibench project.
Update dfsioe.conf to version 6.0, add dfsioe's prepare and mapreduce script.

### CODE REVIEW

File: bin/workloads/micro/dfsioe/mapreduce/hadoop.conf [hunk 1]
```diff
@@ -0,0 +1,2 @@
+
+# override for hadoop related workload confs
```
Comment (carsonwang) [inline bin/workloads/micro/dfsioe/mapreduce/hadoop.conf]: This file is no longer used. Can you remove it?

File: bin/workloads/micro/dfsioe/mapreduce/bin/run-read.sh [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    -tputFile ${WORKLOAD_RESULT_FOLDER}/throughput_read.csv 
+
+END_TIME=`timestamp`
+export HADOOP_OPTS="$OLD_HADOOP_OPTS"
+stop-monitor $MONITOR_PID
+
+gen_report ${START_TIME} ${END_TIME} ${SIZE}
+show_bannar finish
+leave_bench
+
```
Comment (carsonwang) [inline bin/workloads/micro/dfsioe/mapreduce/bin/run-read.sh]: Can you rename the outside folder `mapreduce` to `hadoop` as we are using hadoop now?

File: bin/workloads/micro/dfsioe/prepare/prepare.sh [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+#    -Dtest.build.data=${INPUT_HDFS}                                                 \
+#    -write -skipAnalyze -nrFiles ${RD_NUM_OF_FILES} -fileSize ${RD_FILE_SIZE} -bufferSize 4096 
+
+
+#END_TIME=`timestamp`
+#stop-monitor $MONITOR_PID
+
+#show_bannar finish
+#leave_bench
+
```
Comment (carsonwang) [inline bin/workloads/micro/dfsioe/prepare/prepare.sh]: Can you remove the above comment lines?

File: bin/workloads/micro/dfsioe/prepare/prepare.sh.template [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    -write -skipAnalyze -nrFiles ${RD_NUM_OF_FILES} -fileSize ${RD_FILE_SIZE} -bufferSize 4096 
+
+
+END_TIME=`timestamp`
+stop-monitor $MONITOR_PID
+
+show_bannar finish
+leave_bench
+
+
```
Comment (carsonwang) [inline bin/workloads/micro/dfsioe/prepare/prepare.sh.template]: Let's remove this file as well.

File: bin/workloads/micro/dfsioe/spark/java/bin/run.sh [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+root_dir=${current_dir}/../../../../../../../
+workload_config=${root_dir}/conf/workloads/micro/dfsioe.conf
+. "${root_dir}/bin/functions/load-bench-config.sh"
+
+enter_bench JavaSparkDFSIOE ${workload_config}
+show_bannar start
+echo -e "${On_Blue}Spark DFSIOE not implemented yet, nothing to do.${Color_Off}"
+show_bannar finish
+leave_bench
+
```
Comment (carsonwang) [inline bin/workloads/micro/dfsioe/spark/java/bin/run.sh]: Remove this file

File: bin/workloads/micro/dfsioe/spark/java/java.conf [hunk 6]
```diff
@@ -0,0 +1,2 @@
+
+# override for java related workload confs
```
Comment (carsonwang) [inline bin/workloads/micro/dfsioe/spark/java/java.conf]: Remove this as well.

File: bin/workloads/micro/dfsioe/spark/python/bin/run.sh [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+root_dir=${current_dir}/../../../../../../../
+workload_config=${root_dir}/conf/workloads/micro/dfsioe.conf
+. "${root_dir}/bin/functions/load-bench-config.sh"
+
+enter_bench PythonSparkDFSIOE ${workload_config}
+show_bannar start
+echo -e "${On_Blue}Spark DFSIOE not implemented yet, nothing to do.${Color_Off}"
+show_bannar finish
+leave_bench
+
```
Comment (carsonwang) [inline bin/workloads/micro/dfsioe/spark/python/bin/run.sh]: Remove this file as well.

File: bin/workloads/micro/dfsioe/spark/python/python.conf [hunk 8]
```diff
@@ -0,0 +1,2 @@
+
+# override for python related workload confs
```
Comment (carsonwang) [inline bin/workloads/micro/dfsioe/spark/python/python.conf]: Remove this file as well.

File: bin/workloads/micro/dfsioe/spark/scala/bin/run.sh [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+root_dir=${current_dir}/../../../../../../../
+workload_config=${root_dir}/conf/workloads/micro/dfsioe.conf
+. "${root_dir}/bin/functions/load-bench-config.sh"
+
+enter_bench ScalaSparkDFSIOE ${workload_config}
+show_bannar start
+echo -e "${On_Blue}Spark DFSIOE not implemented yet, nothing to do.${Color_Off}"
+show_bannar finish
+leave_bench
+
```
Comment (carsonwang) [inline bin/workloads/micro/dfsioe/spark/scala/bin/run.sh]: Let's remove the spark directory directly!

File: conf/workloads/micro/dfsioe.conf [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+hibench.dfsioe.read.number_of_files     ${hibench.dfsioe.${hibench.scale.profile}.read.number_of_files}
+hibench.dfsioe.read.file_size           ${hibench.dfsioe.${hibench.scale.profile}.read.file_size}
+hibench.dfsioe.write.number_of_files    ${hibench.dfsioe.${hibench.scale.profile}.write.number_of_files}
+hibench.dfsioe.write.file_size          ${hibench.dfsioe.${hibench.scale.profile}.write.file_size}
+
+# export for shell script
+hibench.dfsioe.base.hdfs                ${hibench.hdfs.data.dir}/Dfsioe
+hibench.workload.input                  ${hibench.dfsioe.base.hdfs}/Input
+hibench.workload.output                 ${hibench.dfsioe.base.hdfs}/Output
+
```
Comment (carsonwang) [inline conf/workloads/micro/dfsioe.conf]: Can you remove `hibench.dfsioe.base.hdfs` and change the above three lines to two:
hibench.workload.input                  ${hibench.hdfs.data.dir}/Dfsioe/Input
hibench.workload.output               ${hibench.hdfs.data.dir}/Dfsioe/Output

**Comment by Moonlit-Sailor** [comment]
move scripts from bin/ to hadoop/, and delete bin hadoop/bin/ directory

**Comment by carsonwang** [comment]
Thanks @Moonlit-Sailor for the work!

**Pull request merged**